stko/calculators/xtb_calculators.py

```diff
@@ -0,0 +1,420 @@
+        self.homo_lumo_orbitals = {}
+        self.qonly_dipole_moments = {}
+        self.full_dipole_moments = {}
+        self.qonly_quadrupole_moments = {}
+        self.qdip_quadrupole_moments = {}
+        self.full_quadrupole_moments = {}
+        self.total_free_energies = {}
+        self.frequencies = {}
+
+    def _get_properties(self, mol, output_file):
+        """
+        Extracts properties from a GFN-xTB energy calculation.
+
+        Parameters
+        ----------
+        mol : :class:`.Molecule`
+            The :class:`.Molecule` whose energy was calculated.
+
+        output_file : :class: `str`
+            Name of the output file with xTB results.
+
+        Returns
+        -------
+        None : :class:`NoneType`
+
+        """
+        # Get properties from output_file.
+        xtbext = XTBExtractor(output_file=output_file)
+
+        self.total_energies[mol] = xtbext.total_energy
+        self.homo_lumo_gaps[mol] = xtbext.homo_lumo_gap
+        self.fermi_levels[mol] = xtbext.fermi_level
+        self.homo_lumo_orbitals[mol] = xtbext.homo_lumo_occ
+        self.qonly_dipole_moments[mol] = xtbext.qonly_dipole_moment
+        self.full_dipole_moments[mol] = xtbext.full_dipole_moment
+        self.qonly_quadrupole_moments[mol] = \
+            xtbext.qonly_quadrupole_moment
+        self.qdip_quadrupole_moments[mol] = \
+            xtbext.qdip_quadrupole_moment
+        self.full_quadrupole_moments[mol] = \
+            xtbext.full_quadrupole_moment
+        if self._calculate_free_energy:
+            self.total_free_energies[mol] = xtbext.total_free_energy
+            self.frequencies[mol] = xtbext.frequencies
+
+    def _run_xtb(self, xyz, out_file):
+        """
+        Runs GFN-xTB.
+
+        Parameters
+        ----------
+        xyz : :class:`str`
+            The name of the input structure ``.xyz`` file.
+
+        out_file : :class:`str`
+            The name of output file with xTB results.
+
+        Returns
+        -------
+        None : :class:`NoneType`
+
+        """
+
+        # Modify the memory limit.
+        if self._unlimited_memory:
+            memory = 'ulimit -s unlimited ;'
+        else:
+            memory = ''
+
+        if self._solvent is not None:
+            solvent = f'--gbsa {self._solvent} {self._solvent_grid}'
+        else:
+            solvent = ''
+
+        if self._calculate_free_energy:
+            calc_type = '--hess'
+        else:
+            calc_type = ''
+
+        cmd = (
+            f'{memory} {self._xtb_path} '
+            f'{xyz} --gfn {self._gfn_version} '
+            f'{calc_type} --parallel {self._num_cores} '
+            f'--etemp {self._electronic_temperature} '
+            f'{solvent} --chrg {self._charge} '
+            f'--uhf {self._num_unpaired_electrons}'
+        )
+
+        with open(out_file, 'w') as f:
+            # Note that sp.call will hold the program until completion
+            # of the calculation.
+            sp.call(
```

Comment:
I think sp.call is being depreciated. Probably best to use sp.run instead

Reply:
I think they have identical arguments, but sp.run handles timeouts better, if they are to be later implemented

Reply:
Gonna make this an issue and merge because I will make changes to this in the near future anyway.